token.go

```diff
@@ -202,7 +217,99 @@ func (t *tokenRing) String() string {
 	return string(buf.Bytes())
 }
 
-func (t *tokenRing) GetHostForToken(token token) (host *HostInfo, endToken token) {
+// Tokens returns the token range corresponding to the primary replica.
+// The elements are sorted by token ascending.
+// The range for a given item starts after preceding range and ends with the token at the current position.
+// The end token is part of the range.
+// The lowest (i.e. index 0) range wraps around the ring (its preceding range is the one with the largest index).
+// You can obtain the owner host/vnode of the range by calling HostForToken with the end token.
+//
+// The following example constructs one TOKEN-based query for each token range:
+//	func buildTokenQueries(s *Session, t *TokenRing) []*Query {
+//		tokens := t.Tokens()
+//		if len(tokens) == 0 {
+//			return nil
+//		}
+//
+//		// If there's a single token (practically impossible), there is no need for a token range query
+//		// because all the tokens are assigned to the same host/vnode anyway
+//		if len(tokens) == 1 {
+//			return []*Query{
+//				s.Query("SELECT * FROM my_table"),
+//			}
+//		}
+//
+//		queries := make([]*Query, 0, len(tokens)+1)
+//		for i, token := range tokens {
+//			var query *Query
+//			if i == 0 {
+//				// The first range contains all the tokens less than or equal to tokens[0]
+//				query = s.Query("SELECT * FROM my_table WHERE TOKEN(id) <= ?", token)
+//			} else {
+//				// Tokens in intermediate ranges are between (tokens[i-1], tokens[i]]
+//				prevToken := tokens[i-1]
+//				query = s.Query("SELECT * FROM my_table WHERE TOKEN(id) > ? AND TOKEN(id) <= ?", prevToken, token)
+//			}
+//			queries = append(queries, query)
+//		}
+//
+//		// Add a final range containing all the tokens greater than the last token,
+//		// to complete the wraparound to tokens[0].
+//		queries = append(queries,
+//			s.Query("SELECT * FROM my_table WHERE TOKEN(id) > ?", tokens[len(tokens)-1]),
+//		)
+//		return queries
+//	}
+func (t *TokenRing) Tokens() []Token {
+	if len(t.tokens) == 0 {
+		return nil
+	}
+
+	ranges := make([]Token, len(t.tokens))
+	for i := range t.tokens {
+		ranges[i] = t.tokens[i].token
+	}
+
+	return ranges
+}
+
+func buildTokenQueries(s *Session, t *TokenRing) []*Query {
```

Comment:
Please remove it, it's unused and the same code is already commented on above as an example.